test/test_framework.py

```diff
@@ -284,48 +284,50 @@ def test_conflicting_event_attributes(self):
         class MyEvent(EventBase):
             pass
 
-        event = Event(MyEvent)
+        event = MyEvent
 
         class MyEvents(EventsBase):
             foo = event
 
+        class MySubEvents(MyEvents):
+            qux = event
+
         class MyNotifier(Object):
             on = MyEvents()
+            on_sub = MySubEvents()
             bar = event
 
         class MyObserver(Object):
             def __init__(self, parent, key):
                 super().__init__(parent, key)
                 self.seen = []
 
-            def on_foo(self, event): self.seen.append(f"on_foo:{event.handle.kind}")
-            def on_bar(self, event): self.seen.append(f"on_bar:{event.handle.kind}")
+            def on_foo(self, event):
+                self.seen.append(f"on_foo:{event.handle.kind}")
+
+            def on_bar(self, event):
+                self.seen.append(f"on_bar:{event.handle.kind}")
+
+            def on_qux(self, event):
+                self.seen.append(f"on_qux:{event.handle.kind}")
 
         pub = MyNotifier(framework, "1")
         obs = MyObserver(framework, "1")
 
+        assert pub.on.foo.emitter_type != pub.on_sub.qux.emitter_type
+        assert pub.on.foo.emitter_type != pub.bar.emitter_type
+        assert pub.on.foo.event_kind != pub.on_sub.qux.event_kind
+        assert pub.on.foo.event_kind != pub.bar.event_kind
+
         framework.observe(pub.on.foo, obs)
         framework.observe(pub.bar, obs)
+        framework.observe(pub.on_sub.qux, obs)
 
         pub.on.foo.emit()
         pub.bar.emit()
+        pub.on_sub.qux.emit()
 
-        self.assertEqual(obs.seen, ["on_foo:foo", "on_bar:bar"])
-
-        # The case where the same value is part of the same hierarchy is completely
-        # unsupported, though, and is detected to prevent awkward bugs.
```

Comment:
As a side-effect of switching from the introspection-based code to `__set_name__`, it no longer is an issue if an event is reused within the same type hierarchy.  However, as mentioned in the comments earlier in this test, and in my review comment above, we might prevent any reuse whatsoever of event classes.  I'm not completely sure if I lean more one way or the other.